src/structure-link/structure-link.ts

```diff
@@ -0,0 +1,36 @@
+import { HttpClientProxy } from '../utils/http-client-proxy';
+import { StructureLink as StructureLinkDto } from './dto/structure-link';
+
+export class StructureLink {
+  private url: string;
+
+  constructor(host: string) {
+    this.url = `http://edtech-structure-link-service.${host}/structures/links`;
+  }
+
+  public async listStructureLinks(
+    options?: Partial<StructureLinkDto>,
+  ): Promise<StructureLinkDto[]> {
+    return await new HttpClientProxy().get<StructureLinkDto[]>(this.url, {
+      params: options,
+    });
+  }
+
+  public async getOne(id: string): Promise<StructureLinkDto> {
+    const response = await new HttpClientProxy().get<StructureLinkDto[]>(
+      this.url,
+      {
+        params: { id },
+      },
+    );
+
+    if (!response.length) {
+      throw {
```

Comment:
Use HttpClientProxyError for consistency

Reply:
This casting is the best I can do, as `HttpClientProxyError` is only as interface.